api/turing/internal/testutils/validation.go

```diff
@@ -16,21 +15,7 @@ func CompareObjects(actual interface{}, expected interface{}) error {
 		allowUnexportedOn = reflect.ValueOf(actual).Elem().Interface()
 	}
 	if !cmp.Equal(actual, expected, cmp.AllowUnexported(allowUnexportedOn)) {
-		actualString := fmt.Sprintf("%+v", actual)
-		expectedString := fmt.Sprintf("%+v", expected)
-
-		// Attempt to encode values to JSON, for logging
-		jsonActual, err := json.Marshal(actual)
-		if err == nil {
-			actualString = string(jsonActual)
-		}
-		jsonExpected, err := json.Marshal(expected)
-		if err == nil {
-			expectedString = string(jsonExpected)
-		}
-
-		return fmt.Errorf("Did not get expected configuration.\nEXPECTED:\n%v\nGOT:\n%v",
-			expectedString, actualString)
+		return fmt.Errorf(cmp.Diff(actual, expected, cmp.AllowUnexported(allowUnexportedOn)))
```

Comment:
I simplified all these steps because the error message it generates isn't precise/specific/useful enough (e.g. for pointing out the presence of null fields especially if those fields get omitted when converted to json) and replaced it with the output of the `Diff` method of the sample `cmp` package.